spring-integration-cassandra/src/main/java/org/springframework/integration/cassandra/outbound/CassandraMessageHandler.java

```diff
@@ -69,13 +70,23 @@
 	private MessageProcessor<Statement> statementProcessor;
 
 	private EvaluationContext evaluationContext;
+	
+	private String query;
+	
 
 	public CassandraMessageHandler(CassandraOperations cassandraTemplate) {
-		this(cassandraTemplate, Type.INSERT);
+		this(cassandraTemplate, OperationType.INSERT);
 	}
+	
 
-	public CassandraMessageHandler(CassandraOperations cassandraTemplate, CassandraMessageHandler.Type queryType) {
-		Assert.notNull(cassandraTemplate, "'cassandraTemplate' must not be null.");
+	public void setQueryType(OperationType queryType) {
+		this.queryType = queryType;
+	}
```

Comment:
I don't like this setter. It allows to change the behavior of the component. That is dangerous.
So, let's come back just only to the ctor initialization for this option!